validate-deployment-readiness

```diff
@@ -62,20 +67,26 @@ function getDeployedHash(target) {
         method: 'GET'
       },
       res => {
+        const isSuccessCode = res.statusCode === 200;
+
+        if (!isSuccessCode) {
+          reject('HTTP_ERROR');
+        }
+
         res.on('data', d => response = response.concat(d));
         res.on('end', () => {
           const responseJson = JSON.parse(response.join(''));
 
           if (responseJson.status === 'ok') {
             resolve(responseJson.lastMigrationHash);
           } else {
-            reject(responseJson);
+            reject('SERVER_NOT_OK');
           }
-        })
+        });
       }
-    )
+    );
 
-    req.on('error', reject);
+    req.on('error', (error) => reject('REQUEST_ERROR'));
```

Comment:
Could just `reject(error)` here? Anything to make debugging easier.

Reply:
I agree, but I was trying to keep from having a union type here. Having it always be a string here is helpful nice, but maybe also `console.error` the error here so it doesn't get swallowed?

Reply:
Whatever you prefer is cool with me! Personally I'd rather have error objects across the board & potentially subclassing Error for certain cases.

I *believe* throwing other stuff breaks stack traces, I'm not totally sure though since I don't think I've ever done it, but it seems like potentially [some kind of best practice](https://eslint.org/docs/rules/no-throw-literal) to not do it.

This script is so simple though that I really don't have any strong feelings at all, feel free to merge as-is and we can have a deeper conversation on the subject somewhere where it'll make a bigger difference (e.g. cala-api where we subclass error a whole bunch).

Reply:
Yeah, I think my instinct here is that we're not *actually* throwing just rejecting a promise, but I know the `async`/`await` syntax makes it feel more like traditional throwing. I'll just use an `Error` here since I feel like this first script will be enough of an example to future work that it's worth getting as close to the style we would want as possible.